src/App.tsx

```diff
@@ -42,54 +64,105 @@ export function getReorderedGoods(
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  reverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortBy = (sortType: SortType) => {
+    this.setState(() => ({
+      sortType,
+    }));
+  };
+
+  reset = () => {
+    this.setState(() => ({
+      isReversed: false,
+      sortType: SortType.NONE,
+    }));
+  };
+
+  render() {
+    const reorderedGoods = getReorderedGoods(goodsFromServer, this.state);
+
+    const visibleReset = this.state.isReversed === false
+      && this.state.sortType === SortType.NONE;
```

Comment:
it strange that you use `this ` to get state values here & next line in code you destructuring this values